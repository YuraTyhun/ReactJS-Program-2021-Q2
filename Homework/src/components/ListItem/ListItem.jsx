import './ListItem.scss';

import React from 'react';
import PropTypes from 'prop-types';

const ListItem = ({ title }) => (
    <li className="list-item">{title}</li>
);

ListItem.propTypes = {
    title: PropTypes.string
};

export default ListItem;

