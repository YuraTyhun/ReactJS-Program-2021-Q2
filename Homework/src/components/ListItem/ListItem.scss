.list-item {
    color: #ffffff;
    padding: 0 0 14px;
    margin-right: 20px;
    font-size: 18px;
    cursor: pointer;
    
    &:nth-child(1) {
        border-bottom: 2px solid #f65261;
    }

    &:hover {
        color: #f65261;
    }
}
