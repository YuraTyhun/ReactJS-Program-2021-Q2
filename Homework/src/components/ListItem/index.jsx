import React from 'react';
import PropTypes from 'prop-types';

import './ListItem.scss';

const ListItem = ({ title }) => (
    <li className="list-item">{title}</li>
);

ListItem.propTypes = {
    title: PropTypes.string
};

export default ListItem;

