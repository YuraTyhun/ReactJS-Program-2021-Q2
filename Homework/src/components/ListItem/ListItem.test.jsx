import React from 'react';
import renderer from 'react-test-renderer';

import ListItem from './index';

it('Filter item renders correctly', () => {
  const filterItem = renderer
    .create(<ListItem title="Comedy" isActive setFilter={() => {}} />)
    .toJSON();
  expect(filterItem).toMatchSnapshot();
});
