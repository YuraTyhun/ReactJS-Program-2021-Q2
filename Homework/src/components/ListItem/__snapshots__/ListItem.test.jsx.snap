// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter item renders correctly 1`] = `
<li
  className="list-item active"
  onClick={[Function]}
>
  Comedy
</li>
`;
